include/quic/connection.hpp

```diff
@@ -92,9 +92,15 @@ namespace oxen::quic
 
         void io_ready();
 
+        /// Returns a pointer to the owning Server, if this is a Server connection, nullptr
+        /// otherwise.
         Server* server();
+        const Server* server() const;
 
+        /// Returns a pointer to the owning Client, if this is a Client connection, nullptr
+        /// otherwise.
         Client* client();
+        const Client* client() const;
```

Comment:
would it be better to do an operator overload for the client and server classes s.t. we can pass them as const?